InvenTree/plugin/builtin/barcodes/digikey_barcode.py

```diff
@@ -0,0 +1,60 @@
+"""The DigiKeyBarcodePlugin matches DigiKey barcodes to supplier parts."""
+
+import logging
+
+from django.utils.translation import gettext_lazy as _
+
+from company.models import SupplierPart
+from plugin import InvenTreePlugin
+from plugin.mixins import BarcodeMixin
+from .barcode2d import parse_ecia_barcode2d
+
+logger = logging.getLogger('inventree')
+
+
+class DigiKeyBarcodePlugin(BarcodeMixin, InvenTreePlugin):
+    """BarcodePlugin for matching DigiKey barcodes."""
+
+    NAME = "DigiKeyBarcode"
+    TITLE = _("DigiKey Barcodes")
+    DESCRIPTION = _("Provides support for scanning DigiKey barcodes")
+    VERSION = "1.0.0"
+    AUTHOR = _("InvenTree contributors")
+
+    def scan(self, barcode_data):
+        """Process a barcode to determine if it is a DigiKey barcode."""
+
+        if not (barcode_fields := parse_ecia_barcode2d(barcode_data)):
+            return
+
+        if not (sku := barcode_fields.get("supplier_part_number")):
+            return
+
+        supplier_parts = SupplierPart.objects.filter(SKU__iexact=sku)
```

Comment:
There's a fair bit of shared code here between the Digikey and Mouser plugins. I think it would be worth creating a top-level class which implements the generic ecia barcode functionality with helper methods such as `get_supplier_part()`?

Reply:
To me this is one of thoses cases where applying DRY would actually lead to less maintainable, more convoluted code. The `get_supplier_part()` function would only be like 3 lines and would have to live in yet another file probably. Not worth it imo.

The one thing I would like to abstract out is "adding the additional vendor barcode data to the response", this part:
```py
if quantity := barcode_fields.get("quantity"):
    try:
        data["quantity"] = int(quantity)
    except ValueError:
        logger.warning(
            f"Failed to parse quantity '{quantity}' with "
            f"MouserBarcodePlugin plugin"
        )

if order_number := barcode_fields.get("purchase_order_number"):
    data["order_number"] = order_number
```
I'm just still not set on the exact format of this, like I mentioned [here](https://github.com/inventree/InvenTree/pull/5509#issuecomment-1707411820) (I think I'd prefer the second example there).